operator/v1alpha1/types_image_content_source_policy.go

```diff
@@ -65,3 +73,34 @@ type RepositoryDigestMirrors struct {
 	// +optional
 	Mirrors []string `json:"mirrors"`
 }
+
+// RepositoryMirrors holds cluster-wide information about how to handle mirrors in the registries config.
+// Note: this is different from the RepositoryDigestMirrors that the mirrors only
+// work when pulling the images that are referenced by their digests.
+type RepositoryMirrors struct {
+	// source is the repository that users refer to, e.g. in image pull specifications.
+	// i.e. using the following format
+	// host[:port]
+	// host[:port]/namespace[/namespace…]
+	// host[:port]/namespace[/namespace…]/repo
+	// +required
+	Source string `json:"source"`
+	// If true, the mirrors only work when pulling the images that are referenced by their digests.
+	// Pulling images by tag can potentially yield different images, depending on which endpoint
+	// we pull from. Forcing digest-pulls for mirrors avoids that issue.
+	// +optional
+	AllowMirrorByTags bool `json:"allowMirrorByTags,omitempty"`
+	// mirrors is one or more repositories that may also contain the same images.
+	// i.e. using the following format
+	// host[:port]
+	// host[:port]/namespace[/namespace…]
+	// host[:port]/namespace[/namespace…]/repo
+	// The order of mirrors in this list is treated as the user's desired priority, while source
+	// is by default considered lower priority than all mirrors. Other cluster configuration,
+	// including (but not limited to) other repositoryMirrors objects,
+	// may impact the exact order mirrors are contacted in, or some mirrors may be contacted
+	// in parallel, so this should be considered a preference rather than a guarantee of ordering.
+	// +optional
+	// +listType=set
+	Mirrors []string `json:"mirrors,omitempty"`
```

Comment:
specify that it is one or more.

Reply:
why is it optional if it must be one or more?

Reply:
I copied the comments from above [Line69 ](https://github.com/openshift/api/pull/874/files#diff-3e682d59341b1b7653cfe3febd52d932b396b2d8bdf62fc196e2be24c6f0e660R69).
in the comment, I can point out the Mirrors is allowed to be an empty slice.
or,  the Mirrors should be changed to ` // +required`?

Reply:
> why is it optional if it must be one or more?

I copied the comments from above RepositoryDigestMirrors [Line69 ](https://github.com/openshift/api/pull/874/files#diff-3e682d59341b1b7653cfe3febd52d932b396b2d8bdf62fc196e2be24c6f0e660R69). They should stay as the same implementation (compared with the `RepositoryDigestMirrors`, `RepositoryMirrors` only has new field `allowMirrorByTags`). 
The `mirrors` field is optional in the CRD.
 If not specified, no mirror is set. https://github.com/openshift/runtime-utils/blob/master/pkg/registries/registries.go#L56
If specified only `mirrors:` in the CRD, one or more mirror should be specified. Use only  `mirrors` will cause error `spec.repositoryDigestMirrors.mirrors: Invalid value: "null": spec.repositoryDigestMirrors.mirrors in body must be of type array: "null"`
```
spec:
  repositoryDigestMirrors:
  - mirrors:
    source: docker.io/library/alpine
```

Reply:
> why is it optional if it must be one or more?

@umohnani8 I copied the current API implementation since the API I added could use the same field. Any change I should make for the comment?

Reply:
The comment should probably be, ```mirrors can be zero-to-many host/port pairs. A zero length field will not set a miror. A maximum of '???' of mirrors are supported```....

We need to define the maximum number of mirrors.

Reply:
Update the comment.
This api change is about allowing using mirror by tags. The number of mirrors hasn't been discussed.

Reply:
@sttts PTAL.

Reply:
Formally fine. But the comment reads very strange:
```
	// mirrors is one or many repositories that may also contain the same images.
	// A zero length Mirrors will not set a mirror.
```
Is it one or many or is zero allowed? And do you distinguish between undefined and empty list?

Reply:
Is an empty string a valid mirror? This needs specification of the values.